### DISCUSSION TITLE
Change 3D material

### DISCUSSION THREAD

**Discussion by uncleheng** [google/filament#6759] [General]
Platform:
Android
Could you please change the material (texture) of gltf and glb models?
I saw the code of replacing filamesh material in demo, but did not see the information about replacing gltf and glb
If it can be changed, could you tell me how to change it? Please tell me the code example
thank you！！！

**Comment by JonaNorman** [comment]
I just made this
```
MaterialInstance[] materialInstances = filamentAsset.getMaterialInstances();
        for (MaterialInstance materialInstance : materialInstances) {
            if (materialInstance.getName().equals("key")){
                materialInstance.setParameter("baseColorMap", texture, new TextureSampler());
            }
        }
```
